Prisma 7 no longer allowed as peer dependency since 1.4.7

### Is this suited for github?

- [x] Yes, this is suited for github

### To Reproduce

Since 1.4.7  prisma is pinned as a peer dependency to `^5.22.0`. Happened in #6654. With this, it is no longer possible to use newer prisma versions (6 or 7) with better-auth. 

```
npm error While resolving: project@1.0.0
npm error Found: prisma@7.1.0
npm error node_modules/prisma
npm error   peerOptional prisma@"*" from @prisma/client@7.1.0
npm error   node_modules/@prisma/client
npm error     peerOptional overridden @prisma/client@"7.1.0" (was "^5.22.0") from better-auth@1.4.7
npm error     node_modules/better-auth
npm error       better-auth@"*" from the root project
npm error     peerOptional overridden @prisma/client@"7.1.0" (was "*") from drizzle-orm@0.41.0
npm error     node_modules/drizzle-orm
npm error       peerOptional drizzle-orm@"^0.41.0" from better-auth@1.4.7
npm error       node_modules/better-auth
npm error         better-auth@"*" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional prisma@"^5.22.0" from better-auth@1.4.7
npm error node_modules/better-auth
npm error   better-auth@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```


The [prisma adapter docs](https://www.better-auth.com/docs/adapters/prisma) mentions prisma 7, which creates the expectation that v7 is supported. 

### Current vs. Expected behavior

Support prisma 7

### What version of Better Auth are you using?

1.4.7

### System info

```bash
no output from `npx @better-auth/cli info --json`
```

### Which area(s) are affected? (Select all that apply)

Package

### Auth config (if applicable)

```typescript

```

### Additional context

Repro:
```
~/projects/abc $ npm init
This utility will walk you through creating a package.json file.
It only covers the most common items, and tries to guess sensible defaults.

See `npm help init` for definitive documentation on these fields
and exactly what they do.

Use `npm install <pkg>` afterwards to install a package and
save it as a dependency in the package.json file.

Press ^C at any time to quit.
package name: (abc)
version: (1.0.0)
description:
entry point: (index.js)
test command:
git repository:
keywords:
author:
license: (ISC)
type: (commonjs)
About to write to /home/sijmen/projects/abc/package.json:

{
  "name": "abc",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "type": "commonjs"
}


Is this OK? (yes)

~/projects/abc $ npm install --save prisma @prisma/client

added 93 packages, and audited 94 packages in 9s

11 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
~/projects/abc $ npm install --save better-auth
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: abc@1.0.0
npm error Found: prisma@7.1.0
npm error node_modules/prisma
npm error   prisma@"^7.1.0" from the root project
npm error   peerOptional prisma@"*" from @prisma/client@7.1.0
npm error   node_modules/@prisma/client
npm error     @prisma/client@"^7.1.0" from the root project
npm error     peerOptional @prisma/client@"*" from drizzle-orm@0.41.0
npm error     node_modules/drizzle-orm
npm error       peerOptional drizzle-orm@"^0.41.0" from better-auth@1.4.7
npm error       node_modules/better-auth
npm error         better-auth@"*" from the root project
npm error
npm error Could not resolve dependency:
npm error peerOptional prisma@"^5.22.0" from better-auth@1.4.7
npm error node_modules/better-auth
npm error   better-auth@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error
npm error For a full report see:
npm error /home/sijmen/.npm/_logs/2025-12-14T10_36_52_740Z-eresolve-report.txt
npm error A complete log of this run can be found in: /home/sijmen/.npm/_logs/2025-12-14T10_36_52_740Z-debug-0.log
~/projects/abc $ cat package.json
{
  "name": "abc",
  "version": "1.0.0",
  "description": "",
  "license": "ISC",
  "author": "",
  "type": "commonjs",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@prisma/client": "^7.1.0",
    "prisma": "^7.1.0"
  }
}
```

--- a/packages/better-auth/package.json
+++ b/packages/better-auth/package.json
@@ -492,7 +492,7 @@
   },
   "peerDependencies": {
     "@lynx-js/react": "*",
-    "@prisma/client": "^5.22.0",
+    "@prisma/client": "^5.0.0 || ^6.0.0 || ^7.0.0",
     "@sveltejs/kit": "^2.0.0",
     "@tanstack/react-start": "^1.0.0",
     "better-sqlite3": "^12.4.1",
@@ -502,7 +502,7 @@
     "mysql2": "^3.14.4",
     "next": "^14.0.0 || ^15.0.0 || ^16.0.0",
     "pg": "^8.16.3",
-    "prisma": "^5.22.0",
+    "prisma": "^5.0.0 || ^6.0.0 || ^7.0.0",
     "react": "^18.0.0 || ^19.0.0",
     "react-dom": "^18.0.0 || ^19.0.0",
     "solid-js": "^1.0.0",

